house/house.go

```diff
@@ -0,0 +1,364 @@
+package house
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/beaujr/nmap_prometheus/assistant"
+	"github.com/beaujr/nmap_prometheus/notifications"
+	pb "github.com/beaujr/nmap_prometheus/proto"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/robfig/cron/v3"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+)
+
+type networkId struct {
+	Ip  string `json:"ip",yaml:"ip"`
+	Mac string `json:"mac",yaml:"mac"`
+}
+
+type device struct {
+	Id       networkId `json:"id",yaml:"id"`
+	LastSeen int64     `json:"lastSeen",yaml:"lastSeen"`
+	Away     bool      `json:"away",yaml:"away"`
+	Name     string    `json:"name",yaml:"name"`
+	Person   bool      `json:"person",yaml:"person"`
+	Command  string    `json:"command",yaml:"command"`
+	Smart    bool      `json:"smart",yaml:"smart"`
+}
+
+//IOT iotDevices
+var timeAwaySeconds = flag.Int64("timeout", 300, "")
+var houseDevices = []*device{}
+var iotDevices = []*device{}
+var syncStatusWithGA = time.Hour.Seconds()
+var metrics map[string]prometheus.Gauge
+var debug = flag.Bool("debug", false, "Debug mode")
+
+var (
+	people_home = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "home_detector_people_home",
+		Help: "The total number of houseDevices at home",
+	})
+)
+
+type HomeManager interface {
+	adjustLights(lightGroup string, brightness string) error
+	DeviceDetectState(phone device) int64
+	DeviceManager() error
+	isDeviceOn(iot *device) (bool, error)
+	isHouseEmpty() bool
+	httpHealthCheck(url string) bool
+	IotDeviceManager(iotDevice *device) error
+	iotStatusManager() error
+	recordMetrics()
+}
+type Server struct {
+	pb.UnimplementedHomeDetectorServer
+}
+
+func readConfig(filename string) []*device {
+	// Open our yamlFile
+	yamlFile, err := os.Open(filename)
+	// if we os.Open returns an error then handle it
+	if err != nil {
+		fmt.Println(err)
+	}
+	log.Println(fmt.Sprintf("Successfully Opened: %s", filename))
+	// defer the closing of our yamlFile so that we can parse it later on
+	defer yamlFile.Close()
+
+	byteValue, _ := ioutil.ReadAll(yamlFile)
+
+	var result []*device
+	err = yaml.Unmarshal(byteValue, &result)
+
+	return result
+}
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+func writeConfig(filename string) error {
+	d1, err := yaml.Marshal(append(houseDevices, iotDevices...))
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(filename, d1, 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func NewServer() HomeManager {
```

Comment:
warning: exported function NewServer should have comment or be unexported (golint)